gnomad/utils/sparse_mt.py

```diff
@@ -917,32 +918,85 @@ def compute_coverage_stats(
     It needs to be keyed with the same keys as `mt`, typically either `locus` or `locus, alleles`.
     The `reference_ht` can e.g. be created using `get_reference_ht`
 
-    :param mt: Input sparse MT
+    :param mtds: Input sparse MT or VDS
     :param reference_ht: Input reference HT
+    :param interval_ht: Table containing intervals to filter to
     :param coverage_over_x_bins: List of boundaries for computing samples over X
     :return: Table with per-base coverage stats
     """
-    n_samples = mt.count_cols()
-    print(f"Computing coverage stats on {n_samples} samples.")
+    is_vds = isinstance(mtds, hl.vds.VariantDataset)
+    if is_vds:
+        n_samples = mtds.variant_data.count_cols()
+    else:
+        n_samples = mtds.count_cols()
+    logging.info("Computing coverage stats on %d n_samplessamples.", n_samples)
+
+    # Filter datasets to interval list
+    if interval_ht is not None:
+        reference_ht = hl.filter_intervals(
+            reference_ht, interval_ht["interval"].collect()
+        )
+
+        if is_vds:
+            mtds = hl.vds.filter_intervals(
+                vds=mtds, intervals=interval_ht, split_reference_blocks=True
+            )
+        else:
+            raise NotImplementedError(
+                "Filtering to an interval list for a sparse Matrix Table is currently"
+                " not supported."
+            )
 
     # Create an outer join with the reference Table
-    mt = mt.select_entries("END", "DP").select_cols().select_rows()
-    col_key_fields = list(mt.col_key)
-    t = mt._localize_entries("__entries", "__cols")
-    t = t.join(reference_ht.key_by(*mt.row_key).select(_in_ref=True), how="outer")
-    t = t.annotate(
-        __entries=hl.or_else(
-            t.__entries,
-            hl.range(n_samples).map(lambda x: hl.null(t.__entries.dtype.element_type)),
+    def join_with_ref(mt: hl.MatrixTable) -> hl.MatrixTable:
```

Comment:
OK, after thinking through this, I think we want this section to be something like this:

```
    def join_with_ref(mt: hl.MatrixTable) -> hl.MatrixTable:
        """
        Outer join MatrixTable with reference Table to add 'in_ref' annotation
        indicating whether a given position is found in the reference Table and filter
        to sites contained in the reference.

        :param mt: Input MatrixTable.
        :return: MatrixTable with 'in_ref' annotation added.
        """
        keep_entries = ["DP"]
        if "END" in mt.entry:
            keep_entries.append("END")
        if "LGT" in mt.entry:
            keep_entries.append("LGT")
        if "GT" in mt.entry:
            keep_entries.append("GT")
        mt_col_key_fields = list(mt.col_key)
        mt_row_key_fields = list(mt.row_key)
        t = mt.select_entries(*keep_entries).select_cols().select_rows()
        t = t._localize_entries("__entries", "__cols")
        t = t.key_by(*row_key_fields).join(
            reference_ht.key_by(*row_key_fields).select(_in_ref=True),
            how="outer",
        ).key_by(*mt_row_key_fields)
        t = t.annotate(
            __entries=hl.or_else(
                t.__entries,
                hl.range(n_samples).map(
                    lambda x: hl.missing(t.__entries.dtype.element_type)
                ),
            )
        )

        return t._unlocalize_entries("__entries", "__cols", mt_col_key_fields)

    if is_vds:
        mtds = hl.vds.VariantDataset(
            mtds.reference_data.select_entries("END", "DP").select_cols().select_rows(),
            join_with_ref(mtds.variant_data),
        )

        # Densify
        mt = hl.vds.to_dense_mt(mtds)
    else:
        mtds = join_with_ref(mtds)
        # Densify
        mt = hl.experimental.densify(mtds)

    mt = mt.filter_rows(mt._in_ref)
```

Where `row_key_fields: List[str] = ["locus"]` should be added to the parameter list

See my notebook at http://localhost:8123/notebooks/gnomad-julia/gnomad_v4/test_coverage_pr.ipynb

Reply:
why keep GT/LGT and why does vds.reference_data no longer get passed to join_with_ref?

Reply:
The VDS wasn't densifying without the GT/LGT. The variant data will already be fully extended to all ref HT loci and add the row ref info that is needed for filtering after the densify